aws-datastore/src/main/java/com/amplifyframework/datastore/AWSDataStorePlugin.java

```diff
@@ -82,13 +83,16 @@
     // overrides provided via the userProvidedConfiguration
     private DataStoreConfiguration pluginConfiguration;
 
+    private final AuthModeStrategyType authModeStrategy;
```

Comment:
why do we need this instance variable for the datastore plugin? shouldn't it reside in the API plugin so that datastore uses whichever strategy that the API category uses?

Reply:
TLDR; From a customer's perspective, multiauth is a DataStore feature. To enable multiauth, the customer has to set that option during creation of the DataStore plugin. That setting controls whether the sync engine submits the API requests with the multiauth flag or not.

A little background...
Initially, the intent was to have the feature be at the API level. However, after some discussion it was decided that multiauth was a DataStore feature mainly due to the nature of the interaction between DataStore and API. When invoking API directly, developers have the option to pass in an authorization type so it didn't make sense to have that option available. 

As far as implementation goes, it made more sense (at least for native) to implement the bulk of the feature on the API side mainly because of the complexity it would have added to the sync engine. We would have had to potentially create a wrapper operation object to keep track of retries and then try to decide whether to bubble up the error or try a different auth type. 

To limit the use of that feature to just the sync engine, only the AppSync request factory under DataStore sets that parameter on the request. The implementation is such that we can make it available for direct API calls as well in the future, but as part of this launch, the feature only applies to DataStore.

Reply:
hmm interesting. i can't say i agree with this, but it seems that we aren't really at a stage to redesign the approach